### PR TITLE
Updates Docker and windows distribution build files to reflect to changes in admin user setup

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
As a follow-up of #4250, this PR updates docker and windows distribution install script to adapt to changes in demo configuration setup.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
@prudhvigodithi @rishabh6788 Mind reviewing this?

File: docker/release/config/opensearch/opensearch-docker-entrypoint.sh
Comment (peterzhuamazon) [inline docker/release/config/opensearch/opensearch-docker-entrypoint.sh:40]: Similar to #4250.

File: scripts/opensearch-onetime-setup.sh
Comment (peterzhuamazon) [inline scripts/opensearch-onetime-setup.sh:27]: Same as above.

File: scripts/startup/zip/windows/opensearch-windows-install.bat
Comment (peterzhuamazon) [inline scripts/startup/zip/windows/opensearch-windows-install.bat:21]: Same as above, but need to check whether windows has a different way.

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]

File: docker/release/config/opensearch/opensearch-docker-entrypoint.sh
Comment (DarshitChanpura) [reply docker/release/config/opensearch/opensearch-docker-entrypoint.sh:40]: based on offline discussion we are going to avoid adding `set -e` to exit the setup scripts, instead we will use `|| exit 1` to hard stop the script in case of any failures with demo script

File: docker/release/dockercomposefiles/docker-compose-2.x.yml
Comment (DarshitChanpura) [inline docker/release/dockercomposefiles/docker-compose-2.x.yml:39]: these changes the default docker-compose files for 2.x line to now include an additional env variable (to be passed by user) to set the admin password.

This will be accompanied by the change in download instructions on the project-website ([file to change](https://github.com/opensearch-project/project-website/blob/main/_includes/downloads/opensearch-docker.markdown?plain=1)) to reflect the new requirement as an additional step.

**Comment by DarshitChanpura** [comment]
@peterzhuamazon @prudhvigodithi @rishabh6788 Can I please get more reviews?

**Comment by peterzhuamazon** [comment]
Also need @prudhvigodithi to confirm on @DarshitChanpura comment above regarding keeping the exit 1 for docker and tar, but remove for deb and rpm.

**Comment by DarshitChanpura** [comment]
> Need to change these as well if you plan to change docker-compose: https://opensearch.org/samples/docker-compose.yml https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/#sample-docker-composeyml

Yes, I have PRs out to to address both:
- https://github.com/opensearch-project/project-website/pull/2526
- https://github.com/opensearch-project/documentation-website/pull/6058

**Comment by DarshitChanpura** [comment]
> Also need @prudhvigodithi to confirm on @DarshitChanpura comment above regarding keeping the exit 1 for docker and tar, but remove for deb and rpm.

@peterzhuamazon RPM and DEB setup already has `set -e` and hence no need for `|| exit 1`. There are no changes to RPM/DEB setup in this PR.

**Comment by DarshitChanpura** [comment]
@peterzhuamazon Could I get re-review on this PR?

**Comment by prudhvigodithi** [comment]
The same https://github.com/opensearch-project/opensearch-build/pull/4250#issuecomment-1887731477 goes here as well.

**Comment by prudhvigodithi** [comment]
@DarshitChanpura can you as `echo -e "This is a line with a new line.\nThis is the second line."` to print statements in a new line. We dont need 3 echo lines.

```
This is a line with a new line.
This is the second line
```

**Comment by DarshitChanpura** [comment]
Similar to https://github.com/opensearch-project/opensearch-build/pull/4250#issuecomment-1887763045, I've updated docker related scripts with printf. There is no echo -e or printf for windows from what I understand

**Comment by DarshitChanpura** [comment]
@peterzhuamazon @prudhvigodithi @rishabh6788 Could I get some more reviews?

**Review by prudhvigodithi — APPROVED** [review]

File: docker/release/config/opensearch/opensearch-docker-entrypoint.sh
Comment (peterzhuamazon) [reply docker/release/config/opensearch/opensearch-docker-entrypoint.sh:38]: `echo -e` should be more posix compliant than `printf`.

**Comment by DarshitChanpura** [comment]
@peterzhuamazon Is there anything blocking this PR still?

**Comment by DarshitChanpura** [comment]
CI task seemed to have failed due to connectivity issues, and re-run should solve it.
```
Warning: Failed to download action 'https://api.github.com/repos/codecov/codecov-action/tarball/f32b3a3741e1053eb607407145bc9619351dc93b'. Error: Operation timed out (api.github.com:443)
Warning: Back off 21.499 seconds before retry.
Warning: Failed to download action 'https://api.github.com/repos/codecov/codecov-action/tarball/f32b3a3741e1053eb607407145bc9619[35](https://github.com/opensearch-project/opensearch-build/actions/runs/7546189581/job/20543417266?pr=4274#step:1:39)1dc93b'. Error: Operation timed out (api.github.com:443)
Warning: Back off 24.923 seconds before retry.
Error: nodename nor servname provided, or not known (api.github.com:443)
```

File: docker/release/dockercomposefiles/docker-compose-2.x.yml
Comment (peterzhuamazon) [reply docker/release/dockercomposefiles/docker-compose-2.x.yml:39]: We need to add a comment for this line, so that people knows it is only for 2.12.0 and up.
Also, the documentation and the download page needs to be tweaked with the same changes as well.

https://opensearch.org/docs/2.11/install-and-configure/install-opensearch/docker/
https://opensearch.org/downloads.html

**Comment by DarshitChanpura** [comment]
@peterzhuamazon @prudhvigodithi @rishabh6788 Could I get some more reviews?

File: docker/release/dockercomposefiles/docker-compose-2.x.yml
Comment (prudhvigodithi) [reply docker/release/dockercomposefiles/docker-compose-2.x.yml:39]: Hey @DarshitChanpura please take a look at this.

Comment (DarshitChanpura) [reply docker/release/dockercomposefiles/docker-compose-2.x.yml:39]: I added it above, and following the convention, I didn't add it second time as no other settings that have comments on the first one have comments here.

Comment (DarshitChanpura) [reply docker/release/dockercomposefiles/docker-compose-2.x.yml:39]: @prudhvigodithi @peterzhuamazon ^^

**Comment by DarshitChanpura** [comment]
@peterzhuamazon Could I get your reviews on this? Also could you or a maintainer trigger re-run for Lint Checker as it timed out during the latest run.

**Pull request merged**
